mirobo/airpurifier.py

```diff
@@ -0,0 +1,182 @@
+from .device import Device
+from typing import Any, Dict
+import enum
+
+
+class AirPurifier(Device):
+    """Main class representing the air purifier."""
+
+    def status(self):
+        """Retrieve properties."""
+
+        # A few more properties:
+        properties = ['power', 'aqi', 'humidity', 'temp_dec',
+                      'mode', 'led', 'led_b', 'buzzer', 'child_lock',
+                      'limit_hum', 'trans_level', 'bright',
+                      'favorite_level', 'filter1_life', 'act_det',
+                      'f1_hour_used', 'use_time', 'motor1_speed']
+
+        values = self.send(
+            "get_prop",
+            properties
+        )
+        return AirPurifierStatus(dict(zip(properties, values)))
+
+    def set_mode(self, mode: OperationMode):
+        """Set mode."""
+
+        # auto, silent, favorite, medium, high, strong, idle
+        return self.send("set_mode", [mode.value])
+
+    def set_favorite_level(self, level: int):
+        """Set favorite level."""
+
+        # Set the favorite level used when the mode is `favorite`,
+        # should be  between 0 and 16.
+        return self.send("favorite_level", [level])  # 0 ... 16
+
+    def set_led_brightness(self, brightness: int):
+        """Set led brightness."""
+
+        # bright: 0, dim: 1, off: 2
+        return self.send("set_led_b", [brightness])
+
+    def set_led(self, led: bool):
+        """Turn led on/off."""
+
```

Comment:
No need for extra white-spaces after the docstring, odd that hound did not yell about that...